Docker Etwlogs driver causes docker service crash

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#reporting-other-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- Docker Support Forums - https://forums.docker.com
- IRC - irc.freenode.net #docker channel
- Post a question on StackOverflow, using the Docker tag

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

I am running docker with log-driver `etwlogs`. Seems this driver cause docker crash

**Steps to reproduce the issue:**
1.  ```docker run --log-driver etwlogs someimage```

by inspect
```
        "HostConfig": {
            "LogConfig": {
                "Type": "etwlogs",
                "Config": {}
            },
```

2. image writes into stdout/stderr

3. docker service crashed occasionally

**Describe the results you received:**

Found in `panic.log`

```
panic: syscall: string with NUL passed to StringToUTF16

goroutine 6641 [running]:
panic(0x127ada0, 0xc0425f4cc0)
	/usr/local/go/src/runtime/panic.go:500 +0x1af
syscall.StringToUTF16(0xc04226e000, 0x9bf, 0xc04226e000, 0x9bf, 0x9bf)
	/usr/local/go/src/syscall/syscall_windows.go:31 +0xc2
syscall.StringToUTF16Ptr(0xc04226e000, 0x9bf, 0x10)
	/usr/local/go/src/syscall/syscall_windows.go:66 +0x3c
github.com/docker/docker/daemon/logger/etwlogs.callEventWriteString(0xc04226e000, 0x9bf, 0xc04226e000, 0x9bf)
	/go/src/github.com/docker/docker/daemon/logger/etwlogs/etwlogs_windows.go:164 +0x96
github.com/docker/docker/daemon/logger/etwlogs.(*etwLogs).Log(0xc04291ce00, 0xc043165d10, 0xc042c56a20, 0xc042925e00)
	/go/src/github.com/docker/docker/daemon/logger/etwlogs/etwlogs_windows.go:73 +0x154
github.com/docker/docker/daemon/logger.(*Copier).copySrc(0xc04291d080, 0x14c8beb, 0x6, 0x22fc438, 0xc04291ce40)
	/go/src/github.com/docker/docker/daemon/logger/copier.go:88 +0x3de
created by github.com/docker/docker/daemon/logger.(*Copier).Run
	/go/src/github.com/docker/docker/daemon/logger/copier.go:41 +0x11f

```

**Describe the results you expected:**

No Crash

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64

Server:
 Version:      1.12.2-cs2-ws-beta
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   050b611
 Built:        Tue Oct 11 02:35:40 2016
 OS/Arch:      windows/amd64
```

**Output of `docker info`:**

```
Containers: 6
 Running: 0
 Paused: 0
 Stopped: 6
Images: 3
Server Version: 1.12.2-cs2-ws-beta
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: nat null overlay
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.576.amd64fre.rs1_release_inmarket.161208-2252)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 28 GiB
Name: vme1f7682836
ID: G6NE:BKJN:TETO:LHFW:7STN:T2XA:NE2R:H4Z2:PSAO:R6BB:HNPJ:RGH3
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Azure

--- a/daemon/logger/etwlogs/etwlogs_windows.go
+++ b/daemon/logger/etwlogs/etwlogs_windows.go
@@ -146,7 +146,13 @@ func callEventRegister() error {
 }
 
 func callEventWriteString(message string) error {
-	ret, _, _ := procEventWriteString.Call(uintptr(providerHandle), 0, 0, uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr(message))))
+	utf16message, err := syscall.UTF16FromString(message)
+
+	if err != nil {
+		return err
+	}
+
+	ret, _, _ := procEventWriteString.Call(uintptr(providerHandle), 0, 0, uintptr(unsafe.Pointer(&utf16message[0])))
 	if ret != win32CallSuccess {
 		errorMessage := fmt.Sprintf("ETWLogs provider failed to log message. Error: %d", ret)
 		logrus.Error(errorMessage)

